Error copying file "/CMakeFiles/libui.dir/windows/resources.rc.obj" to "D:/libui/build/out/libui.res".

Windows 10 64-bit
gcc version 5.2.0 (x86_64-posix-seh-rev0, Built by MinGW-W64 project)

```
D:\libui\build [master...origin/master +1 ~0 -0 !]> cmake -G "MinGW Makefiles" -DBUILD_SHARED_LIBS=OFF ..
-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- Check for working C compiler: C:/mingw/mingw64/bin/gcc.exe
-- Check for working C compiler: C:/mingw/mingw64/bin/gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/mingw/mingw64/bin/g++.exe
-- Check for working CXX compiler: C:/mingw/mingw64/bin/g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: D:/libui/build

D:\libui\build [master...origin/master +1 ~0 -0 !]> mingw32-make
Scanning dependencies of target libui
[  1%] Building C object CMakeFiles/libui.dir/common/areaevents.c.obj
[  3%] Building C object CMakeFiles/libui.dir/common/control.c.obj
[  5%] Building C object CMakeFiles/libui.dir/common/debug.c.obj
[  6%] Building C object CMakeFiles/libui.dir/common/matrix.c.obj
[  8%] Building C object CMakeFiles/libui.dir/common/shouldquit.c.obj
[ 10%] Building C object CMakeFiles/libui.dir/common/userbugs.c.obj
[ 12%] Building CXX object CMakeFiles/libui.dir/windows/alloc.cpp.obj
[ 13%] Building CXX object CMakeFiles/libui.dir/windows/area.cpp.obj
[ 15%] Building CXX object CMakeFiles/libui.dir/windows/areadraw.cpp.obj
[ 17%] Building CXX object CMakeFiles/libui.dir/windows/areaevents.cpp.obj
[ 18%] Building CXX object CMakeFiles/libui.dir/windows/areascroll.cpp.obj
D:\libui\windows\areascroll.cpp:137:13: warning: 'void hscrollto(uiArea*, intmax_t)' defined but not used [-Wunused-func
tion]
 static void hscrollto(uiArea *a, intmax_t pos)
             ^
D:\libui\windows\areascroll.cpp:182:13: warning: 'void vscrollto(uiArea*, intmax_t)' defined but not used [-Wunused-func
tion]
 static void vscrollto(uiArea *a, intmax_t pos)
             ^
[ 20%] Building CXX object CMakeFiles/libui.dir/windows/areautil.cpp.obj
[ 22%] Building CXX object CMakeFiles/libui.dir/windows/box.cpp.obj
D:\libui\windows\box.cpp: In function 'void boxRelayout(uiBox*)':
D:\libui\windows\box.cpp:86:5: warning: suggest explicit braces to avoid ambiguous 'else' [-Wparentheses]
  if (nStretchy != 0)
     ^
D:\libui\windows\box.cpp:38:12: warning: unused variable 'i' [-Wunused-variable]
  uintmax_t i;
            ^
D:\libui\windows\box.cpp:40:19: warning: unused variable 'd' [-Wunused-variable]
  uiWindowsSizing *d;
                   ^
D:\libui\windows\box.cpp: In function 'void uiBoxMinimumSize(uiWindowsControl*, intmax_t*, intmax_t*)':
D:\libui\windows\box.cpp:160:12: warning: unused variable 'i' [-Wunused-variable]
  uintmax_t i;
            ^
D:\libui\windows\box.cpp:162:18: warning: unused variable 'sizing' [-Wunused-variable]
  uiWindowsSizing sizing;
                  ^
D:\libui\windows\box.cpp: In function 'void boxArrangeChildren(uiBox*)':
D:\libui\windows\box.cpp:233:12: warning: unused variable 'i' [-Wunused-variable]
  uintmax_t i;
            ^
[ 24%] Building CXX object CMakeFiles/libui.dir/windows/button.cpp.obj
[ 25%] Building CXX object CMakeFiles/libui.dir/windows/checkbox.cpp.obj
[ 27%] Building CXX object CMakeFiles/libui.dir/windows/colorbutton.cpp.obj
[ 29%] Building CXX object CMakeFiles/libui.dir/windows/colordialog.cpp.obj
D:\libui\windows\colordialog.cpp: In function 'void hsv2RGB(double, double, double, double*, double*, double*)':
D:\libui\windows\colordialog.cpp:90:9: warning: unused variable 'c1' [-Wunused-variable]
  double c1, c2;
         ^
D:\libui\windows\colordialog.cpp:90:13: warning: unused variable 'c2' [-Wunused-variable]
  double c1, c2;
             ^
D:\libui\windows\colordialog.cpp: In function 'HWND__* replaceWithD2DScratch(HWND, int, SUBCLASSPROC, void*)':
D:\libui\windows\colordialog.cpp:868:43: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  return newD2DScratch(parent, &r, (HMENU) id, subproc, (DWORD_PTR) data);
                                           ^
[ 31%] Building CXX object CMakeFiles/libui.dir/windows/combobox.cpp.obj
[ 32%] Building CXX object CMakeFiles/libui.dir/windows/container.cpp.obj
[ 34%] Building CXX object CMakeFiles/libui.dir/windows/control.cpp.obj
[ 36%] Building CXX object CMakeFiles/libui.dir/windows/d2dscratch.cpp.obj
[ 37%] Building CXX object CMakeFiles/libui.dir/windows/datetimepicker.cpp.obj
[ 39%] Building CXX object CMakeFiles/libui.dir/windows/debug.cpp.obj
D:\libui\windows\debug.cpp: In function 'HRESULT _logLastError(const WCHAR*, const WCHAR*, const WCHAR*, const WCHAR*)':
D:\libui\windows\debug.cpp:22:13: warning: ISO C++ forbids converting a string constant to 'WCHAR* {aka wchar_t*}' [-Wpe
dantic]
   formatted = L"\n";
             ^
D:\libui\windows\debug.cpp: In function 'HRESULT _logHRESULT(const WCHAR*, const WCHAR*, const WCHAR*, const WCHAR*, HRE
SULT)':
D:\libui\windows\debug.cpp:49:13: warning: ISO C++ forbids converting a string constant to 'WCHAR* {aka wchar_t*}' [-Wpe
dantic]
   formatted = L"\n";
             ^
[ 41%] Building CXX object CMakeFiles/libui.dir/windows/draw.cpp.obj
[ 43%] Building CXX object CMakeFiles/libui.dir/windows/drawmatrix.cpp.obj
[ 44%] Building CXX object CMakeFiles/libui.dir/windows/drawpath.cpp.obj
[ 46%] Building CXX object CMakeFiles/libui.dir/windows/drawtext.cpp.obj
D:\libui\windows\drawtext.cpp: In function 'uiDrawTextFont* mkTextFont(IDWriteFont*, BOOL, WCHAR*, BOOL, double)':
D:\libui\windows\drawtext.cpp:64:10: warning: unused variable 'hr' [-Wunused-variable]
  HRESULT hr;
          ^
D:\libui\windows\drawtext.cpp: In function 'void dwriteAttrToAttr(dwriteAttr*)':
D:\libui\windows\drawtext.cpp:177:15: warning: variable 'curindex' set but not used [-Wunused-but-set-variable]
  int curdiff, curindex;
               ^
D:\libui\windows\drawtext.cpp: In function 'void uiDrawText(uiDrawContext*, double, double, uiDrawTextLayout*)':
D:\libui\windows\drawtext.cpp:502:10: warning: unused variable 'hr' [-Wunused-variable]
  HRESULT hr;
          ^
[ 48%] Building CXX object CMakeFiles/libui.dir/windows/dwrite.cpp.obj
[ 50%] Building CXX object CMakeFiles/libui.dir/windows/editablecombo.cpp.obj
[ 51%] Building CXX object CMakeFiles/libui.dir/windows/entry.cpp.obj
D:\libui\windows\entry.cpp: In function 'uiEntry* uiNewSearchEntry()':
D:\libui\windows\entry.cpp:126:10: warning: variable 'hr' set but not used [-Wunused-but-set-variable]
  HRESULT hr;
          ^
[ 53%] Building CXX object CMakeFiles/libui.dir/windows/events.cpp.obj
D:\libui\windows\events.cpp: In function 'void issueWM_WININICHANGE(WPARAM, LPARAM)':
D:\libui\windows\events.cpp:147:23: warning: unused variable 'ch' [-Wunused-variable]
  struct wininichange *ch;
                       ^
[ 55%] Building CXX object CMakeFiles/libui.dir/windows/fontbutton.cpp.obj
[ 56%] Building CXX object CMakeFiles/libui.dir/windows/fontdialog.cpp.obj
D:\libui\windows\fontdialog.cpp: In function 'void fontDialogDrawSampleText(fontDialog*, ID2D1RenderTarget*)':
D:\libui\windows\fontdialog.cpp:367:10: warning: ISO C++ forbids converting a string constant to 'WCHAR* {aka wchar_t*}'
 [-Wpedantic]
   sample = L"The quick brown fox jumps over the lazy dog.";
          ^
[ 58%] Building CXX object CMakeFiles/libui.dir/windows/graphemes.cpp.obj
[ 60%] Building CXX object CMakeFiles/libui.dir/windows/group.cpp.obj
[ 62%] Building CXX object CMakeFiles/libui.dir/windows/init.cpp.obj
D:\libui\windows\init.cpp: In function 'const char* initerr(const char*, const WCHAR*, DWORD)':
D:\libui\windows\init.cpp:24:10: warning: ISO C++ forbids converting a string constant to 'WCHAR* {aka wchar_t*}' [-Wped
antic]
   sysmsg = L"";
          ^
[ 63%] Building CXX object CMakeFiles/libui.dir/windows/label.cpp.obj
[ 65%] Building CXX object CMakeFiles/libui.dir/windows/main.cpp.obj
[ 67%] Building CXX object CMakeFiles/libui.dir/windows/menu.cpp.obj
[ 68%] Building CXX object CMakeFiles/libui.dir/windows/multilineentry.cpp.obj
[ 70%] Building CXX object CMakeFiles/libui.dir/windows/parent.cpp.obj
[ 72%] Building CXX object CMakeFiles/libui.dir/windows/progressbar.cpp.obj
[ 74%] Building CXX object CMakeFiles/libui.dir/windows/radiobuttons.cpp.obj
D:\libui\windows\radiobuttons.cpp: In function 'BOOL onWM_COMMAND(uiControl*, HWND, WORD, LRESULT*)':
D:\libui\windows\radiobuttons.cpp:21:12: warning: unused variable 'i' [-Wunused-variable]
  uintmax_t i;
            ^
[ 75%] Building CXX object CMakeFiles/libui.dir/windows/separator.cpp.obj
[ 77%] Building CXX object CMakeFiles/libui.dir/windows/sizing.cpp.obj
[ 79%] Building CXX object CMakeFiles/libui.dir/windows/slider.cpp.obj
D:\libui\windows\slider.cpp:29:51: warning: extra ';' [-Wpedantic]
 uiWindowsControlAllDefaultsExceptDestroy(uiSlider);
                                                   ^
[ 81%] Building CXX object CMakeFiles/libui.dir/windows/spinbox.cpp.obj
[ 82%] Building CXX object CMakeFiles/libui.dir/windows/stddialogs.cpp.obj
[ 84%] Building CXX object CMakeFiles/libui.dir/windows/tab.cpp.obj
[ 86%] Building CXX object CMakeFiles/libui.dir/windows/tabpage.cpp.obj
[ 87%] Building CXX object CMakeFiles/libui.dir/windows/text.cpp.obj
[ 89%] Building CXX object CMakeFiles/libui.dir/windows/utf16.cpp.obj
[ 91%] Building CXX object CMakeFiles/libui.dir/windows/utilwin.cpp.obj
[ 93%] Building CXX object CMakeFiles/libui.dir/windows/window.cpp.obj
D:\libui\windows\window.cpp: In function 'void windowRelayout(uiWindow*)':
D:\libui\windows\window.cpp:38:18: warning: unused variable 'sizing' [-Wunused-variable]
  uiWindowsSizing sizing;
                  ^
D:\libui\windows\window.cpp: In function 'void uiWindowMinimumSize(uiWindowsControl*, intmax_t*, intmax_t*)':
D:\libui\windows\window.cpp:227:18: warning: unused variable 'sizing' [-Wunused-variable]
  uiWindowsSizing sizing;
                  ^
[ 94%] Building CXX object CMakeFiles/libui.dir/windows/winpublic.cpp.obj
[ 96%] Building CXX object CMakeFiles/libui.dir/windows/winutil.cpp.obj
[ 98%] Building RC object CMakeFiles/libui.dir/windows/resources.rc.obj
[100%] Linking CXX static library out\libui.a
Copying libui.res
Error copying file "/CMakeFiles/libui.dir/windows/resources.rc.obj" to "D:/libui/build/out/libui.res".
CMakeFiles\libui.dir\build.make:1597: recipe for target 'out/libui.a' failed
mingw32-make[2]: *** [out/libui.a] Error 1
mingw32-make[2]: *** Deleting file 'out/libui.a'
CMakeFiles\Makefile2:66: recipe for target 'CMakeFiles/libui.dir/all' failed
mingw32-make[1]: *** [CMakeFiles/libui.dir/all] Error 2
makefile:82: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
D:\libui\build [master...origin/master +1 ~0 -0 !]>
```

--- a/windows/CMakeLists.txt
+++ b/windows/CMakeLists.txt
@@ -68,16 +68,11 @@ if(NOT BUILD_SHARED_LIBS)
 	set(_LIBUI_STATIC_RES ${CMAKE_ARCHIVE_OUTPUT_DIRECTORY}/libui.res PARENT_SCOPE)
 endif()
 macro(_handle_static)
-	if(MSVC)
-		set(_res_suffix res)
-	else()
-		set(_res_suffix obj)
-	endif()
 	# TODO this full path feels hacky
 	add_custom_command(
 		TARGET libui POST_BUILD
 		COMMAND
-			${CMAKE_COMMAND} -E copy $<TARGET_PROPERTY:libui,BINARY_DIR>/CMakeFiles/libui.dir/windows/resources.rc.${_res_suffix} ${_LIBUI_STATIC_RES}
+			${CMAKE_COMMAND} -E copy $<TARGET_PROPERTY:libui,BINARY_DIR>/CMakeFiles/libui.dir/windows/resources.rc.* ${_LIBUI_STATIC_RES}
 		COMMENT "Copying libui.res")
 	set(_res_suffix)
 endmacro()

